api/model/src/main/java/org/projectnessie/api/v2/params/Transplant.java

```diff
@@ -71,23 +72,35 @@ public interface Transplant extends BaseMergeTransplant {
   @jakarta.validation.constraints.Size(min = 1)
   String getMessage();
 
+  /**
+   * Lists the hashes of commits that should be transplanted into the target branch.
+   *
+   * <p>Since Nessie spec 2.1.1, hashes can be absolute or relative.
+   */
   @NotNull
   @jakarta.validation.constraints.NotNull
   @Size
   @jakarta.validation.constraints.Size(min = 1)
-  List<String> getHashesToTransplant();
+  List<
+          @Pattern(
```

Comment:
I manually validated that this annotation works both with Jaxrs/Weld and Quarkus. 

(For Jaxrs, I added Hibernate Validator recently in #7193, for Quarkus it's already present in quarkus-common.)

The problem is, the bean is being validated by the `@Check` method too, and that method is executed first.

If you remove the `@Check` method, then you would observe how Hibernate Validator validates that:

```
transplantCommitsIntoBranch.transplant.hashesToTransplant[0].<list element>: Hash with optional relative part must consist of either ... [etc.]
```

To be fair, I think we don't need the `@Check` method anymore, and besides, the HV error message is more explicit because it mentions where the problem was found. But maybe keeping it is safer, because we own the validation code. Wdyt?

Reply:
OK but let's do it in a follow-up PR. I fear that some tests could fail, e.g. if they expect an exact error message.

Reply:
#7316